### ISSUE TITLE
mediaPlayer is null

### ISSUE THREAD

**Issue by hoseineghbal** [dooboolab/react-native-audio-recorder-player#27]
### Version of react-native-audio-recorder-player
^2.1.1
### Platforms you faced the error (IOS or Android or both?)
Android
### Expected behavior
define one constant of player like this ::
const audioRecorderPlayer = new AudioRecorderPlayer();`

now recording success with this method ::
startRecord = async () => {
        if (!await this.checkPermission()) {
            console.log('permission problem')
            return;
        }
        var RNFS = require('react-native-fs');
        let dst = RNFS.DocumentDirectoryPath + '/files';
        const path = Platform.select({
            ios: dst + '/hello.m4a',
            android: dst + '/hello.mp4', // should give extra dir name in android. Won't grant permission to the first level of dir.
        });
        const uri = await audioRecorderPlayer.startRecorder(path);
        this.props.startRecording(uri);
        audioRecorderPlayer.addRecordBackListener((e) => {
            this.props.updateRecordingProgress(audioRecorderPlayer.mmssss(Math.floor(e.current_position)));
            return;
        });
        console.log(uri);
    };

after finishing recording stop like this :::
stopRecord = async () => {
        this.props.stopRecording();
        const result = await audioRecorderPlayer.stopRecorder();
        audioRecorderPlayer.removeRecordBackListener();
        console.log(result);

        console.log('wanna upload :: ' + this.props.record.uri)
        const uploadResult = await UploadFile(this.props.record.uri);
        console.log('audio uploadResult => ' + JSON.stringify(uploadResult))
  
        console.log('finish recording in :: ' + this.props.record.uri);
    }

now i wanna play it with this command :: 
plusClick = async () => {
        console.log('plusClick')
        var RNFS = require('react-native-fs');
        let dst = RNFS.DocumentDirectoryPath + '/files';
        const path = Platform.select({
            ios: dst + '/hello.m4a',
            android: dst + '/hello.mp4',
        });
        console.log('startPlayer :: ' + path)
        const msg = await audioRecorderPlayer.startPlayer(path);
        //audioRecorderPlayer.setVolume(1.0);
        console.log(msg);
        audioRecorderPlayer.addPlayBackListener((e) => {
            if (e.current_position === e.duration) {
                console.log('finished and called Stop');
                
                audioRecorderPlayer.stopPlayer().then(() => {

                });
            }
            this.setState({
                currentPositionSec: e.current_position,
                currentDurationSec: e.duration,
                playTime: audioRecorderPlayer.mmssss(Math.floor(e.current_position)),
                duration: audioRecorderPlayer.mmssss(Math.floor(e.duration)),
            });
            return;
        });
    }

first time everythings is ok but second time error shows that ::
`12-11 18:00:37.863 25690 30527 W ReactNativeJS: Error: mediaPlayer is null.
12-11 18:00:37.863 25690 30527 W ReactNativeJS: createErrorFromErrorData@http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:7705:26
12-11 18:00:37.863 25690 30527 W ReactNativeJS: http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:7657:51
12-11 18:00:37.863 25690 30527 W ReactNativeJS: __invokeCallback@http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:8187:23
12-11 18:00:37.863 25690 30527 W ReactNativeJS: http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:7934:34
12-11 18:00:37.863 25690 30527 W ReactNativeJS: __guard@http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:8091:15
12-11 18:00:37.863 25690 30527 W ReactNativeJS: invokeCallbackAndReturnFlushedQueue@http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false:7933:21
12-11 18:00:37.863 25690 30527 W ReactNativeJS: invokeCallbackAndReturnFlushedQueue@[native code]`

<img width="1280" alt="screen shot 2018-12-11 at 6 04 35 pm" src="https://user-images.githubusercontent.com/6056458/49808002-b8006c00-fd70-11e8-8084-f2ba68304b05.png">

**Comment by hyochani** [comment]
@hoseineghbal The warning message I put is to inform you to make code thread-safe. There may be in some situation where you try to play `mediaPlayer` when is already stopped and released. Also please describe your flow and your test scenario so that I can understand better. Thanks.

**Comment by hoseineghbal** [comment]
@hyochans 
thanks for your answer...

i wanna use in a messenger then i need recording and stop recording and sometimes playing recorded sound but all of them with one instance of this library ....

but when recording and playing it second time media player is null and i can not do more ...



and a question is posssible to play an online urlwith thislibrary ?? because i upload recorder sound to server and get url of it then i must play it if user wanna ...

**Comment by hyochans** [comment]
@hoseineghbal Of course, you can play audio from the network. Just put your mp3 or any compatible `uri` in `startPlayer()`. Also, for the first question, try to `stopPlayer` first before `startPlayer` if you face that issue.

**Comment by hoseineghbal** [comment]
@hyochans according to picture of error i try to stop but tell me mediaPlayer is null and promise rejection error ...

then i can not do more actions

**Comment by hyochans** [comment]
Then please wrap the code with try and catch statement to prevent throwing.

**Comment by mahermalhem** [comment]
on START play method try to stop the player before it end by it self       -parseInt(500)
in my code i used 500ms to solve it 
that wont be much long period so no one will notice  

onStartPlay = async () => {
    try {
      console.log('onStartPlay');
      const msg = await audioRecorderPlayer.startPlayer();
      console.log(msg);
      audioRecorderPlayer.addPlayBackListener((e) => {
        
        if (e.current_position === e.duration-parseInt(500)) {
          console.log('finished'+e.current_position);
          audioRecorderPlayer.stopPlayer();
          audioRecorderPlayer.removePlayBackListener();
        }
        //do what u want
        return;
      });
    } catch (error) {
      console.log('onStartPlay'+error);
    }
    
  };

hope IT works with u

**Comment by Sargnec** [comment]
I get this warning when my sound playing is finished and when I ty to seekToPlayer(not works and just gives this warning). And there is one more issue current_position is like delayed when sound finishes cuurent_position is not even close to duration
